### PR TITLE
Update the o365 alert based on azure data

### PR DESCRIPTION
**Pull request by hardikhdholariya**
- Added custom command to find diff between two nested json fields
- Updated policy change alert as per new data source azure

### CODE REVIEW

File: cyences_app_for_splunk/bin/azure_ad_field_extraction.py
Comment (VatsalJagani) [inline cyences_app_for_splunk/bin/azure_ad_field_extraction.py:1]: **field_extraction** name seems bit generic, can we do something like **modified properties formatter** or something?

File: cyences_app_for_splunk/default/commands.conf [hunk 1]
```diff
@@ -81,3 +81,7 @@ filename = cyences_upgrade.py
 chunked = true
 python.version = python3
 
+[azureadfieldextraction]
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/default/commands.conf:84]: field_extraction name seems bit generic, can we do something like **modified properties formatter** or something?

**Review by VatsalJagani — CHANGES_REQUESTED** [review]

File: cyences_app_for_splunk/bin/azure_ad_modified_properties_formatter.py [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+                            + str(diff.final_dict[key][0])
+                            + " ==> "
+                            + str(diff.final_dict[key][1])
+                            + "\n"
+                            + "-------------------------------------------------"
+                            + "\n"
+                        )
+
+                    for field in self.additional_fields:
+                        record["old_"+field] = diff.dict1.get(field)
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/bin/azure_ad_modified_properties_formatter.py:109]: What happens when the field does not exist? It shouldn't generate error.

File: cyences_app_for_splunk/default/savedsearches.conf [hunk 3]
```diff
@@ -2170,23 +2170,26 @@ relation = greater than
...
 | `cs_o365_policy_change_internal_filter` \
-| eval cyences_severity = case( in(Target_DisplayName, `cs_ad_important_policy`) or in(Target_DisplayName, `cs_ad_important_policy_internal`), "medium", true(), "low")  \
-| `cs_o365_policy_change_filter`  \
-| table event_time, cyences_severity, Actor, Command, Target_ObjectID, Target_DisplayName, ResultStatus, ModifiedProperties, MetaData
+| eval cyences_severity = case( in(Target_DisplayName, `cs_ad_important_policy`) or in(Target_DisplayName, `cs_ad_important_policy_internal`), "medium", true(), "low") \
+| eval additional_details = mvzip(key, value, " = ") \
+| table _time, cyences_severity, Actor, Command, Target_ObjectID, Target_DisplayName, displayName, oldValue, newValue, ResultStatus, modified_properties, additional_details \
+| azureadmodifiedpropertiesformatter old_value_field=oldValue new_value_field=newValue property_name_field=displayName field_to_update=modified_properties additional_fields=state \
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/default/savedsearches.conf:2187]: Define severity based on state field change. If state field got changed, severity should be **critical** or **high** depending on number of results daily, if it is not more than 7 times in a week change of state, we can go with critical.

**Review by VatsalJagani — CHANGES_REQUESTED** [review]

File: cyences_app_for_splunk/default/savedsearches.conf (hunk 3 above)
Comment (hardikhdholariya) [reply cyences_app_for_splunk/default/savedsearches.conf:2186]: FYI, this alert scheduled hourly. 
didn't get this point => if it is not more than 7 times in a week change of state, we can go with critical.

Added commit with rest of the changes.

File: cyences_app_for_splunk/default/savedsearches.conf [hunk 4]
```diff
@@ -2181,11 +2181,13 @@ request.ui_dispatch_view = search
...
 | rename initiatedBy.user.userPrincipalName as Actor, activityDisplayName as Command, targetResources{}.displayName as Target_DisplayName, result as ResultStatus, targetResources{}.modifiedProperties{}.displayName as displayName, targetResources{}.modifiedProperties{}.newValue as newValue, targetResources{}.modifiedProperties{}.oldValue as oldValue, additionalDetails{}.key as key, additionalDetails{}.value as value, targetResources{}.id as Target_ObjectID \
 | `cs_o365_policy_change_internal_filter` \
-| eval cyences_severity = case( in(Target_DisplayName, `cs_ad_important_policy`) or in(Target_DisplayName, `cs_ad_important_policy_internal`), "medium", true(), "low") \
 | eval additional_details = mvzip(key, value, " = ") \
 | table _time, cyences_severity, Actor, Command, Target_ObjectID, Target_DisplayName, displayName, oldValue, newValue, ResultStatus, modified_properties, additional_details \
 | azureadmodifiedpropertiesformatter old_value_field=oldValue new_value_field=newValue property_name_field=displayName field_to_update=modified_properties additional_fields=state \
 | fields - displayName oldValue newValue \
+| eventstats count by Command \
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/default/savedsearches.conf:2188]: Do not use count.

**Review by VatsalJagani — APPROVED** [review]

**Pull request merged**